### PR TITLE
lmdb: fix cross-build to iOS + don't suggest unofficial cmake names

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/lmdb/all/conanfile.py [hunk 1]
```diff
@@ -59,11 +61,8 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.libs = ["lmdb"]
-        self.cpp_info.names["cmake_find_package"] = "LMDB"
-        self.cpp_info.names["cmake_find_package_multi"] = "LMDB"
-        self.cpp_info.names["pkg_config"] = "lmdb"
```
Comment (ericLemanissier) [inline recipes/lmdb/all/conanfile.py:64]: most distros ship this file:

- https://packages.ubuntu.com/hirsute/amd64/liblmdb-dev/filelist 
- https://debian.pkgs.org/11/debian-main-amd64/liblmdb-dev_0.9.24-1_amd64.deb.html
- https://alpine.pkgs.org/3.14/alpine-main-aarch64/lmdb-dev-0.9.29-r0.apk.html
- https://fedora.pkgs.org/rawhide/fedora-x86_64/lmdb-devel-0.9.29-1.fc35.x86_64.rpm.html

**Review by ericLemanissier — CHANGES_REQUESTED** [review]

File: recipes/lmdb/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/lmdb/all/conanfile.py:64]: Distro are not a reference for pkg_config or cmake files, but well I can keep pkg_config name if you want.

**Review by uilianries — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Regarding https://github.com/conan-io/conan-center-index/pull/6411/commits/729f9c32ee99a7660559205b4f2014502e06218e, I've seen some people here using `install(TARGETS myLib)` in patches or custom CMakeLists, but it's a feature of CMake 3.14+.

File: recipes/lmdb/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/lmdb/all/conanfile.py:64]: restored in https://github.com/conan-io/conan-center-index/pull/6411/commits/78fd2b5ee214958394f0b011fede8b5316efdaac

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**